contracts/CourtRoom.sol

```diff
@@ -7,144 +7,159 @@ import "./ensabstract.sol";
-  }
-
 }
 
 contract SwearGame is Owned {
 
-	uint256 public amountStaked;
-	uint public rewardCompensation;
-	uint  public registeredPlayersCounter;
-	mapping(address => bool) public registeredPlayers;
-	mapping(address => bytes32[]) public clientsClaimsIds;
+	uint256 public deposit;
+	uint public reward;
+	uint  public playerCount;
+	mapping(address => bool) public players;
+	mapping(address => bytes32[]) public ids;
 
 	CaseContract public caseContract;
-	SampleToken public sampleToken;
-	bytes32 public claimId;
-
-
-	function SwearGame(address _caseContract, address _sampleToken, uint _rewardCompensation) {
-		caseContract = CaseContract(_caseContract);
-		sampleToken = SampleToken(_sampleToken);
-		rewardCompensation = _rewardCompensation;
-		registeredPlayersCounter = 0;
-
-	}
-
-	function deposit(uint256 _depositAmount) onlyOwner payable public returns(bool){
-
-		require(sampleToken.balanceOf(owner) >= _depositAmount);
+	SampleToken public token;
+	bytes32 public id;
 
-		require(sampleToken.transferFrom(owner, address(this), _depositAmount));
 
-		amountStaked += _depositAmount;
-
-		DepositStaked(_depositAmount, amountStaked);
-
-		return true;
+	function SwearGame(address _CaseContract, address _token, uint _reward) {
+		caseContract = CaseContract(_CaseContract);
+		token = SampleToken(_token);
+		reward = _reward;
+		playerCount = 0;
 
 	}
 
+  /* The function without name is the default function that is called whenever anyone sends funds to a contract */
+    function () payable {
+        uint amount = msg.value;
+    		require(token.transferFrom(owner, address(this), amount));
+    		deposit += amount;
+    		DepositStaked(amount, deposit);
+    }
 
-	function makeJudgement(bytes32 _claimId) private returns(bool) {
+	function verdict(bytes32 _id) private returns(bool) {
 
-	  bool claimantCompensated;
+	  bool caseCompensated;
+    bool decision = false;
 
-    require(caseContract.getStatus(_claimId) != 0);
+    require(caseContract.getStatus(_id) != 0);
 		// Somehow come to a resolution...
-		var(plaintiff,valid) = caseContract.getClaim(_claimId);
+		var(plaintiff,beneficiary,blockNumber,sig_v,sig_r,sig_s) = caseContract.getClaim(_id);
 
-		// Case has already been compensated for
-		require(!valid);
+    decision = (validatePromise(beneficiary,blockNumber,sig_v,sig_r,sig_s) && guilty());
 
-	  bool decision = takeDecision();
-
-    caseContract.setClaimValid(_claimId);
-
-    if (decision == false){
-	    claimantCompensated = compensate(plaintiff);
-			caseContract.resolveClaim(_claimId);
+    if (decision == true){
+	    caseCompensated = compensate(plaintiff);
+			caseContract.resolveClaim(_id);
     }
 
-		ClaimResolved(_claimId, plaintiff, rewardCompensation);
-		return claimantCompensated;
+		ClaimResolved(_id, plaintiff, reward);
+		return caseCompensated;
 
 	}
 
-	function compensate(address _claimant) private returns(bool compensated) {
+	function compensate(address __Caseant) private returns(bool compensated) {
```

Comment:
`__Caseant` ?
Should this not be `_beneficiary`? As in th person who opened the claim and needs to be compensated?